### ISSUE TITLE
Unix build "cannot link with the boost thread library" during ./configure

### ISSUE THREAD

**Issue by Seipas** [POV-Ray/povray#54]
When building POV-Ray 3.7-stable, at the step ./configure, I have the following error :

> checking whether pthreads work with -pthread... yes
> checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
> checking if more special flags are required for pthreads... no
> checking for boostlib >= 1.37... yes
> checking whether the Boost::Thread library is available... yes
> checking whether the boost thread library is usable... no
> configure: error: in `/home/seipas/Téléchargements/povray-3.7-stable':
> configure: error: cannot link with the boost thread library

config.log :
```
configure:7100: checking whether pthreads work with -pthread
configure:7183: gcc -o conftest  -pthread   conftest.c   >&5
conftest.c: In function 'main':
conftest.c:33:22: warning: null argument where non-null required (argument 1) [-Wnonnull]
                      pthread_attr_init(0); pthread_cleanup_push(0, 0);
                      ^
conftest.c:34:22: warning: null argument where non-null required (argument 1) [-Wnonnull]
                      pthread_create(0,0,0,0); pthread_cleanup_pop(0);
                      ^
conftest.c:34:22: warning: null argument where non-null required (argument 3) [-Wnonnull]
configure:7183: $? = 0
configure:7192: result: yes
configure:7211: checking for joinable pthread attribute
configure:7226: gcc -o conftest  -pthread   conftest.c   >&5
configure:7226: $? = 0
configure:7232: result: PTHREAD_CREATE_JOINABLE
configure:7242: checking if more special flags are required for pthreads
configure:7249: result: no
configure:7384: checking for boostlib >= 1.37
configure:7438: g++ -c  -pthread  -I/usr/include conftest.cpp >&5
configure:7438: $? = 0
configure:7440: result: yes
configure:7606: checking whether the Boost::Thread library is available
configure:7638: g++ -c -pthread  -pthread  -I/usr/include conftest.cpp >&5
configure:7638: $? = 0
configure:7653: result: yes
configure:7849: checking whether the boost thread library is usable
configure:7869: g++ -o conftest  -pthread  -pthread -I/usr/include  -L/usr/lib conftest.cpp   -pthread  >&5
/tmp/ccFv4p8p.o: In function `__static_initialization_and_destruction_0(int, int)':
conftest.cpp:(.text+0x4f): undefined reference to `boost::system::generic_category()'
conftest.cpp:(.text+0x5b): undefined reference to `boost::system::generic_category()'
conftest.cpp:(.text+0x67): undefined reference to `boost::system::system_category()'
collect2: error: ld returned 1 exit status
configure:7869: $? = 1
configure: program exited with status 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "POV-Ray"
| #define PACKAGE_TARNAME "povray"
| #define PACKAGE_VERSION "3.7.0"
| #define PACKAGE_STRING "POV-Ray 3.7.0"
| #define PACKAGE_BUGREPORT "unix-bugreports-2011@povray.org"
| #define PACKAGE_URL ""
| #define VERSION_BASE "3.7"
| #define DISTRIBUTION_MESSAGE_2 " Seipas Alawalichu <seipas@alawalichu>
```

I have Linux Mint 17.3. `sudo apt-cache showpkg libboost-dev` gives : `1.54.0.1ubuntu1`

**Comment by wfpokorny** [comment]
The POV-Ray 3.7-stable branch has an issue which requires ./configure be called with extra options. Namely : 

./configure COMPILED_BY="..." LIBS="-lboost_system -lboost_thread"

Does this work for you? 

Bill P.

**Comment by Seipas** [comment]
I still add an error because I had to install the package `libboost-all-dev` (I just installed `libboost-dev`).

Now I have the following error :
```
Makefiles
---------
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: error: cannot find input file: `Makefile.in'
```
config.log : `config.status:921: error: cannot find input file: `Makefile.in'`

Is this file supposed to be here or supposed to be builded by ./prebuild.sh or ./configure ?

Thank you for your help.

**Comment by wfpokorny** [comment]
I just pulled stable code anew and I too now get the same error where I do not believe I did when I did my 3.7.0 stable build some time ago. In any case, the following patches - which I picked up from an article from the folks at Amazon Web Services - fixed things for me. In the unix directory : 

sed 's/automake --w/automake --add-missing --w/g' -i prebuild.sh
sed 's/dist-bzip2/dist-bzip2 subdir-objects/g' -i configure.ac
./prebuild.sh

Then back in the povray directory try the ./configure ... 

I believe it is the automake command in the prebuild.sh script which creates the makefile.in files.

Bill P.
